src/com/jdngray77/htmldesigner/backend/BackgroundTask.kt

```diff
@@ -1,25 +1,52 @@
 package com.jdngray77.htmldesigner.backend
 
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
 /**
  * Runs tasks in the background
  */
-object BackgroundTask {
+object BackgroundTask : Subscriber {
+
+    private val threadPool = Executors.newCachedThreadPool()
+
+    init {
+        EventNotifier.subscribe(this, EventType.USER_EXIT)
+    }
+
 
     fun runOnUIThread(runnable: Runnable) {
         TODO()
     }
 
+    
+    /**
+     * Submits a [runnable] task to the for [threadPool] for execution.
+     */
     fun submit(runnable: Runnable) {
```

Comment:
This function should be `@Syncronized`, since multiple threads may be submitting tasks at the same time.

Reply:
Done.